src/nvim/edit.c

```diff
@@ -462,6 +462,8 @@ static void insert_enter(InsertState *s)
     o_lnum = curwin->w_cursor.lnum;
   }
 
+  foldUpdateAll(curwin);
+  foldOpenCursor();
```

Comment:
@Shougo This seems to open folds on any cursor movement now, even without changes. Why is `foldOpenCursor()` needed? Can we remove it?